lib/cylc/gui/gsummary.py

```diff
@@ -280,25 +285,53 @@ def launch_gcylc(host, suite, owner=None):
     """Launch gcylc for a given suite and host."""
     if owner is None:
         owner = user
-    stdout = open(os.devnull, "w")
-    stderr = stdout
-    command = "cylc gui --host=%s --user=%s %s" % (
-                                         host, owner, suite)
-    command = shlex.split(command)
-    subprocess.Popen(command, stdout=stdout, stderr=stderr)
+    args = ["--host=" + host, "--user=" + owner, suite]
+
+    # Get version of suite
+    f_null = open(os.devnull, "w")
+    if cylc.flags.debug:
+        stderr = sys.stderr
+        args = ["--debug"] + args
+    else:
+        stderr = f_null
+    command = ["cylc", "get-suite-version"] + args
+    proc = Popen(command, stdout=PIPE, stderr=stderr)
+    suite_version = proc.communicate()[0].strip()
+    proc.wait()
+
+    # Run correct version of "cylc gui", provided that "admin/cylc-wrapper" is
+    # installed.
+    env = None
+    if suite_version != CYLC_VERSION:
+        env = dict(os.environ)
+        env["CYLC_VERSION"] = suite_version
+    command = ["cylc", "gui"] + args
+    if cylc.flags.debug:
+        stdout = sys.stdout
+        stderr = sys.stderr
+        Popen(command, env=env, stdout=stdout, stderr=stderr)
```

Comment:
The problem is like this:
* `bin/cylc` (this branch) *prepends* `lib/` (this branch) to `sys.path` and `PYTHONPATH`.
* `bin/cylc` (this branch) loads `lib/cylc/__init__.py` (this branch).
* `bin/cylc` (this branch) calls `sys.argv[0] + "-gsummary"`.
* `bin/cylc-gsummary` (this branch) calls `lib/cylc/gui/gsummary.py` (this branch).
* `lib/cylc/gui/gsummary.py` (this branch) calls `bin/cylc` (6.4.0).
* `bin/cylc` (6.4.0) *appends* its `lib/` (6.4.0) to `sys.path`.
* `bin/cylc` (6.4.0) incorrectly loads `lib/cylc/__init__.py` (this branch) instead of `lib/cylc/__init__.py` (6.4.0).
  * (This branch fixes this for future versions.)
* `lib/cylc/__init__.py` (this branch) does not add `${CYLC_DIR}/bin` to `${PATH}`.
* `bin/cylc` (6.4.0) attempts to call `cylc-gsummary` which is not in `${PATH}` (and does not get reported nicely).
  * (This branch fixes this for future versions.)

This change will eliminate this problem for future versions. I guess I ought to do something for backward compatibility.